test/mocks/auth/firebase_auth_mocks.dart

```diff
@@ -13,13 +15,24 @@ class FakeFirebaseAuthPeriodic extends Fake implements FirebaseAuth {
       });
 }
 
-class FakeFirebaseAuthWithError extends Fake implements FirebaseAuth {
+class FakeFirebaseAuthSignInException extends Fake implements FirebaseAuth {
   @override
-  Stream<auth.User> authStateChanges() => Stream.fromIterable([FakeAuthUser()]);
+  Future<UserCredential> signInWithCredential(AuthCredential credential) =>
+      throw FirebaseAuthException(
+          code: 'user-not-found', message: 'test error: cant find user');
+}
 
+class FakeFirebaseAuthThrows extends Fake implements FirebaseAuth {
```

Comment:
The name here is a bit more general so I think some commenting would be really helpful - is that OK?